### ISSUE TITLE
Add map of IAM users to module output

### ISSUE THREAD

**Issue by kriswuollett** [terraform-google-modules/terraform-google-sql-db#455]
### TL;DR

A map of IAM database users should be provided in a database cluster module output so that the actual username is available downstream which is different than the email when it is a service account because its `.gserviceaccount.com` suffix gets trimmed. This output would be useful for downstream Terraform modules, e.g., passing the username to a `google_cloud_run_v2_job` that uses Cloud SQL Proxy via `cloud_sql_instance` volume.

### Terraform Resources

- https://registry.terraform.io/providers/hashicorp/google/latest/docs

### Detailed design

In the `postgres` module for example, the `iam_users` is defined as the following which is also directly used as an output:

https://github.com/terraform-google-modules/terraform-google-sql-db/blob/15298c275e2d531e3379d1c445b2317c99e6363b/modules/postgresql/main.tf#L29

Following that something like the following should be added:

```terraform
  iam_user_accounts = {
    for id, user in local.iam_users id => {
      email         = user.email,
      is_account_sa = user.is_account_sa
      username = user.is_account_sa ? trimsuffix(user.email, ".gserviceaccount.com") : user.email
    }
  }
```

Then to `outputs.tf`:

```terraform
output "iam_users_map" {
  description = "The map of the IAM users with access to the CloudSQL instance"
  value       = local.iam_user_accounts
}
```

### Additional information

This feature would be similar to the map provided by the `terraform-google-modules/service-accounts/google`: 

https://github.com/terraform-google-modules/terraform-google-service-accounts/blob/7c9e85d6d7b16a30da1451c0ce65d0e64016cacb/outputs.tf#L43-L46

This module already asks for an `id` that doesn't appear to be used for anything other than a key for a for expression, so it seemed like the intent would be to support such as the map described, otherwise the database username itself would have been a sufficient key for the `google_sql_user` for expression.
